Projects/b_u585i_iot02a_ntz/Src/ota_pal/ota_pal_stm32u5_ntz.c

```diff
@@ -375,390 +591,565 @@ static HAL_StatusTypeDef prvWriteToFlash( uint32_t destination,
+                        }
+
+                        break;
+
+                    case OTA_PAL_REJECTED:
+                        xOtaStatus = OtaPalSuccess;
+                        break;
+
+                    default:
+                        break;
                 }
 
                 break;
 
-            case OtaImageStateRejected:
+            case OtaImageStateTesting:
 
-                /* Make sure the image is not committed already. A committed image cannot be aborted or rejected. */
-                if( pContext->state != OTA_PAL_IMAGE_STATE_COMMITTED )
-                {
-                    /*
-                     *  Remove the persisted state of image from flash.
-                     */
-                    prvRemoveImageContextFromFlash();
-                    pContext->state = OTA_PAL_IMAGE_STATE_INVALID;
-                    otaStatus = OtaPalSuccess;
-                }
-                else
+                switch( pxContext->xPalState )
                 {
-                    otaStatus = OtaPalRejectFailed;
+                    case OTA_PAL_PENDING_ACTIVATION:
+                    case OTA_PAL_PENDING_SELF_TEST:
+
+                        /* Select target bank to start testing */
+                        if( prvSelectBank( pxContext->ulTargetBank ) == pdTRUE )
+                        {
+                            xOtaStatus = OtaPalSuccess;
+                            pxContext->xPalState = OTA_PAL_PENDING_SELF_TEST;
+                            ( void ) prvWritePalNvContext( pxContext );
+                        }
+
+                        break;
+
+                    default:
+                        break;
                 }
 
                 break;
 
             case OtaImageStateAborted:
 
-                /* Make sure the image is not committed already. A committed image cannot be aborted or rejected. */
-                if( pContext->state != OTA_PAL_IMAGE_STATE_COMMITTED )
-                {
-                    /*
-                     *  Remove the persisted state of image from flash.
-                     */
-                    prvRemoveImageContextFromFlash();
-                    pContext->state = OTA_PAL_IMAGE_STATE_INVALID;
-                    otaStatus = OtaPalSuccess;
-                }
-                else
+                switch( pxContext->xPalState )
                 {
-                    otaStatus = OtaPalAbortFailed;
+                    case OTA_PAL_READY:
+                    case OTA_PAL_REJECTED:
+                    case OTA_PAL_ACCEPTED:
+                        xOtaStatus = OtaPalSuccess;
+                        break;
+
+                    case OTA_PAL_FILE_OPEN:
+                    case OTA_PAL_PENDING_ACTIVATION:
+                    case OTA_PAL_PENDING_SELF_TEST:
+                    case OTA_PAL_NEW_IMAGE_BOOTED:
+                    case OTA_PAL_NEW_IMAGE_WDT_RESET:
+
+                        if( ( prvSelectBank( ulGetOtherBank( pxContext->ulTargetBank ) ) == pdTRUE ) &&
```

Comment:
Calling this function from OtaPal_Abort() to abort an ongoing image download can cause issues here.

Reply:
How so? After an aborted job, the pal is immediately ready to process a new job. Further writes will fail because the file enters the closed state.